test/js/substr.spec.js

```diff
@@ -0,0 +1,239 @@
+/**
+* Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
+* This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2022 Datadog, Inc.
+**/
+
+const { TaintedUtils, taintFormattedString, formatTaintedValue } = require('./util')
+const assert = require('assert')
+
+const rangesTestCases = [
```

Comment:
These test cases could not be shared between tests?

Reply:
At first it was a good idea but although the source string is the same for every flavor the expected result differs :(.